src/Package/Impl/Sql/Publish/SProcGenerator.cs

```diff
@@ -0,0 +1,186 @@
+        /// </summary>
+        internal const string RCodeColumnName = "RCode";
+
+        private const string RCodeTemplate = "_RCODE_";
+        private const string InputQueryTemplate = "_INPUT_QUERY_";
+
+        private readonly ICoreShell _coreShell;
+        private readonly IProjectSystemServices _pss;
+        private readonly IFileSystem _fs;
+
+        public SProcGenerator(ICoreShell coreShell, IProjectSystemServices pss, IFileSystem fs) {
+            _coreShell = coreShell;
+            _pss = pss;
+            _fs = fs;
+        }
+
+        /// <summary>
+        /// Generates SQL scripts for the deployment of R code into SQL database.
+        /// </summary>
+        public void Generate(SqlSProcPublishSettings settings, EnvDTE.Project targetProject) {
+            var targetFolder = Path.Combine(Path.GetDirectoryName(targetProject.FullName), "R\\");
+            if (!_fs.DirectoryExists(targetFolder)) {
+                _fs.CreateDirectory(targetFolder);
+            }
+
+            var targetProjectItem = targetProject.ProjectItems.Item("R") ?? targetProject.ProjectItems.AddFolder("R");
+
+            if (settings.CodePlacement == RCodePlacement.Table) {
+                CreateRCodeTable(settings, targetProject, targetFolder, targetProjectItem);
+                CreatePostDeploymentScript(settings, targetProject, targetFolder, targetProjectItem);
+            }
+            CreateStoredProcedures(settings, targetProject, targetFolder, targetProjectItem);
+        }
+
+        /// <summary>
+        /// Create SQL file that defines table template that will hold R code
+        /// </summary>
+        private void CreateRCodeTable(SqlSProcPublishSettings settings, EnvDTE.Project targetProject, string targetFolder, EnvDTE.ProjectItem targetProjectItem) {
+            var creatTableScriptFile = Path.Combine(targetFolder, CreateRCodeTableScriptName);
+            using (var sw = new StreamWriter(creatTableScriptFile)) {
+                sw.WriteLine(Invariant($"CREATE TABLE {settings.TableName}"));
+                sw.WriteLine("(");
+                sw.WriteLine(Invariant($"[{SProcColumnName}] NVARCHAR(64),"));
+                sw.WriteLine(Invariant($"[{RCodeColumnName}] NVARCHAR(max)"));
+                sw.WriteLine(")");
+            }
+            targetProjectItem.ProjectItems.AddFromFile(creatTableScriptFile);
+        }
+
+        /// <summary>
+        /// Generates SQL post deployment script that pushes R code into a table
+        /// as well as 
+        /// </summary>
+        private void CreatePostDeploymentScript(SqlSProcPublishSettings settings, EnvDTE.Project targetProject, string targetFolder, EnvDTE.ProjectItem targetProjectItem) {
+            var postDeploymentScript = Path.Combine(targetFolder, PostDeploymentScriptName);
+
+            using (var sw = new StreamWriter(postDeploymentScript)) {
+                sw.WriteLine(Invariant($"INSERT INTO {settings.TableName}"));
```

Comment:
Just in case we ever want to extend the schema of that table, specifying field names here would keep the scripts compatible.